Tests/CustomIdGeneratorTest.php

```diff
@@ -0,0 +1,90 @@
+use Doctrine\Common\Annotations\AnnotationReader;
+use Doctrine\ORM\EntityManagerInterface;
+use Fixtures\Bundles\AnnotationsBundle\AnnotationsBundle;
+use Fixtures\Bundles\AnnotationsBundle\Entity\TestCustomIdGeneratorEntity;
+use Symfony\Bundle\FrameworkBundle\DependencyInjection\FrameworkExtension;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\ParameterBag\ParameterBag;
+
+class CustomIdGeneratorTest extends TestCase
+{
+    public static function setUpBeforeClass(): void
+    {
+        if (interface_exists(EntityManagerInterface::class)) {
+            return;
+        }
+
+        self::markTestSkipped('This test requires ORM');
+    }
+
+    /**
+     * https://github.com/doctrine/orm/pull/7953 needed, otherwise ORM classes we define services for trigger deprecations
+     *
+     * @group legacy
+     */
+    public function testRepositoryServiceWiring(): void
+    {
+        $container = new ContainerBuilder(new ParameterBag([
+            'kernel.debug' => false,
+            'kernel.bundles' => ['AnnotationsBundle' => AnnotationsBundle::class],
+            'kernel.cache_dir' => sys_get_temp_dir(),
+            'kernel.environment' => 'test',
+            'kernel.runtime_environment' => '%%env(default:kernel.environment:APP_RUNTIME_ENV)%%',
+            'kernel.build_dir' => __DIR__ . '/../../../../', // src dir
+            'kernel.root_dir' => __DIR__ . '/../../../../', // src dir
+            'kernel.project_dir' => __DIR__ . '/../../../../', // src dir
+            'kernel.bundles_metadata' => [],
+            'kernel.charset' => 'UTF-8',
+            'kernel.container_class' => ContainerBuilder::class,
+            'kernel.secret' => 'test',
+            'container.build_id' => uniqid(),
+            'env(base64:default::SYMFONY_DECRYPTION_SECRET)' => 'foo',
+        ]));
+        $container->set('annotation_reader', new AnnotationReader());
+
+        $extension = new FrameworkExtension();
+        $container->registerExtension($extension);
+        $extension->load(['framework' => []], $container);
+
+        $extension = new DoctrineExtension();
+        $container->registerExtension($extension);
+        $extension->load([
+            [
+                'dbal' => [
+                    'driver' => 'pdo_sqlite',
+                    'charset' => 'UTF8',
+                ],
+                'orm' => [
+                    'mappings' => [
+                        'AnnotationsBundle' => [
+                            'type' => 'annotation',
+                            'dir' => __DIR__ . '/DependencyInjection/Fixtures/Bundles/AnnotationsBundle/Entity',
+                            'prefix' => 'Fixtures\Bundles\AnnotationsBundle\Entity',
+                        ],
+                    ],
+                ],
+            ],
+        ], $container);
+
+        $def = $container->register('my_id_generator', CustomIdGenerator::class)
+            ->setPublic(false);
```

Comment:
Why is this call needed? It should be private by default already.

Reply:
the bundle still supports 3.4, where services are not yet private by default

Reply:
symfony/dependency-injection 3.4 is not supported, so I believe this could indeed be dropped. That said, we do support 3.4 for some other components, but that should not interfere here.